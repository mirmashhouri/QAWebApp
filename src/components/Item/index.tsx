import { memo, useState } from "react";
import { Link } from "react-router-dom";

import { Question } from "../../services/questionService";
import styles from "./styles.module.scss";

interface Props {
  item: Question;
}

const Item = ({ item }: Props): JSX.Element => {
  const [isVisible, setIsVisible] = useState(false);
  const QuestionOnClick = () => {
    setIsVisible(!isVisible);
  };

  return (
    <div className={styles.item}>
      <ul>
        <li key={item.id}>
          <button type="button" onClick={QuestionOnClick}>
            {item.question}
          </button>
          <Link to={`/UserInfo/${item.id}`}> edit </Link>
          <span> | </span>
          <Link to={`/UserInfo/${item.id}`}> delete </Link>
          <div className={`${styles.answer} ${isVisible ? styles.open : ""}`}>
            {item.answer}
          </div>
        </li>
      </ul>
    </div>
  );
};

export default memo(Item);
