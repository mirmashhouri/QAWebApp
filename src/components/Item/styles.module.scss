@import "../../theme/variables";

.item {
  color: $color-white;

  a {
    color: $color-dark-gray;
  }
}
.item ul {
  padding-left: 17px;

  li {
    margin-bottom: 0.5em;
  }
}
.answer {
  display: none;
}
.answer.open {
  display: block;
}
