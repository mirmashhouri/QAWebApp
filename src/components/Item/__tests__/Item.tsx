import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import Item from "..";

describe("<Item />", () => {
  it("renders", () => {
    const tree = render(
      <MemoryRouter>
        <Item
          item={{
            id: 1,
            question: "What is your name?",
            answer: "My name is Abbas",
          }}
        />
      </MemoryRouter>
    ).container.firstChild;

    expect(tree).toMatchSnapshot();
  });
});
