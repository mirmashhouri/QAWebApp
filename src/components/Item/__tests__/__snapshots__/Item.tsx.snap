// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders 1`] = `
<div
  class="item"
>
  <ul>
    <li>
      <button
        type="button"
      >
        What is your name?
      </button>
      <a
        href="/UserInfo/1"
      >
         edit 
      </a>
      <span>
         | 
      </span>
      <a
        href="/UserInfo/1"
      >
         delete 
      </a>
      <div
        class="answer "
      >
        My name is Abbas
      </div>
    </li>
  </ul>
</div>
`;
